### PR TITLE
gdcm: add python bindings

### PR DESCRIPTION
**Pull request by davidchall**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Restore the python bindings to the gdcm package from the [homebrew-science formula](https://github.com/Homebrew/homebrew-science/blob/3c6dd94e5befb05f7d65771eea472e6dd72f0961/gdcm.rb). Needed a bit of tinkering to play nicely with the system python. Also added tests of python bindings.

### CODE REVIEW

**Comment by ilovezfs** [comment]
If it's just building bindings, why do they need to be mutually exclusive? (See `opencv`)

**Comment by davidchall** [comment]
It seems that [opencv](https://github.com/Homebrew/homebrew-core/blob/master/Formula/opencv.rb) can do this because its CMakeLists.txt files explicitly support bindings for both python versions (see [L74-L81](https://github.com/Homebrew/homebrew-core/blob/master/Formula/opencv.rb#L74-L81)). On the other hand, GDCM only supports bindings for a single version of python.

**Comment by ilovezfs** [comment]
You should be able to run cmake 2x (see `snappy`)

File: Formula/gdcm.rb [hunk 1]
```diff
@@ -19,17 +23,39 @@ class Gdcm < Formula
...
-                            "-DGDCM_USE_VTK=ON",
-                            "-DGDCM_USE_SYSTEM_OPENSSL=ON",
-                            *std_cmake_args
-      system "make", "install"
+      if build.without?("python") && build.without?("python3")
+        system "cmake", "..", *args
+        system "make", "install"
+      else
+        Language::Python.each_python(build) do |python, version|
+          python_include = `#{python} -c 'from distutils import sysconfig;print(sysconfig.get_python_inc(True))'`.chomp
```
Comment (ilovezfs) [inline Formula/gdcm.rb]: should use Utils.popen_read

File: Formula/gdcm.rb [hunk 2]
```diff
@@ -19,17 +23,39 @@ class Gdcm < Formula
...
-                            "-DGDCM_BUILD_DOCBOOK_MANPAGES=OFF",
-                            "-DGDCM_USE_VTK=ON",
-                            "-DGDCM_USE_SYSTEM_OPENSSL=ON",
-                            *std_cmake_args
-      system "make", "install"
+      if build.without?("python") && build.without?("python3")
+        system "cmake", "..", *args
+        system "make", "install"
+      else
+        Language::Python.each_python(build) do |python, version|
```
Comment (ilovezfs) [inline Formula/gdcm.rb]: should use python_version so the version variable isn't inadvertently hidden

File: Formula/gdcm.rb [hunk 3]
```diff
@@ -19,17 +23,39 @@ class Gdcm < Formula
...
+          python_include = `#{python} -c 'from distutils import sysconfig;print(sysconfig.get_python_inc(True))'`.chomp
+          python_args = %W[
+            -DGDCM_WRAP_PYTHON=ON
+            -DPYTHON_EXECUTABLE='#{python}'
```
Comment (ilovezfs) [inline Formula/gdcm.rb]: don't think these single quotes or any of the ones below are needed

File: Formula/gdcm.rb [hunk 4]
```diff
@@ -19,17 +23,39 @@ class Gdcm < Formula
...
-                            "-DGDCM_USE_SYSTEM_OPENSSL=ON",
-                            *std_cmake_args
-      system "make", "install"
+      if build.without?("python") && build.without?("python3")
+        system "cmake", "..", *args
+        system "make", "install"
+      else
+        Language::Python.each_python(build) do |python, version|
+          python_include = `#{python} -c 'from distutils import sysconfig;print(sysconfig.get_python_inc(True))'`.chomp
+          python_args = %W[
```
Comment (ilovezfs) [inline Formula/gdcm.rb]: args +=

File: Formula/gdcm.rb [hunk 5]
```diff
@@ -3,14 +3,18 @@ class Gdcm < Formula
...
-  revision 1
+  revision 2
 
   bottle do
     sha256 "3fd15851c093b4e0c2481233268e41614b26dde24aa2c13f9107402ae8bf2ddd" => :high_sierra
     sha256 "4fad6b9761ca2ab4c9b26b418d57ef26f2dd19bebb77a658b5b5aadda0212098" => :sierra
     sha256 "dbfbcaa31ceaedc98985614899601f771e951cbd6c115223082ab73ef111f231" => :el_capitan
   end
 
+  option "without-python", "Build without python2 support"
```
Comment (ilovezfs) [inline Formula/gdcm.rb]: blank line under here

File: Formula/gdcm.rb [hunk 6]
```diff
@@ -19,17 +23,39 @@ class Gdcm < Formula
 
   def install
     ENV.cxx11
+    ENV.append "LDFLAGS", "-undefined dynamic_lookup"
```
Comment (ilovezfs) [inline Formula/gdcm.rb]: would move this down to the python else block

File: Formula/gdcm.rb (hunk 4 above)
Comment (davidchall) [reply Formula/gdcm.rb]: But then `args` will become polluted when bindings are built for multiple python versions.

Comment (ilovezfs) [reply Formula/gdcm.rb]: You're right. Let's call the top variable common_args and then you can do args = common_args + %W[ here.

**Comment by ilovezfs** [comment]
@davidchall please squash this down to one commit :)

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
Thanks for your first contribution to Homebrew, @davidchall! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**